common/include/pcl/impl/point_types.hpp

```diff
@@ -800,57 +718,54 @@ namespace pcl
   PCL_EXPORTS std::ostream& operator << (std::ostream& os, const PointXYZHSV& p);
   struct EIGEN_ALIGN16 PointXYZHSV : public _PointXYZHSV
   {
-    inline PointXYZHSV (const _PointXYZHSV &p)
-    {
-      x = p.x; y = p.y; z = p.z; data[3] = 1.0f;
-      h = p.h; s = p.s; v = p.v;
-    }
+    inline constexpr PointXYZHSV (const _PointXYZHSV &p) : 
+      PointXYZHSV{p.x, p.y, p.z, p.h, p.s, p.v} {}
 
-    inline PointXYZHSV (): PointXYZHSV (0.f, 0.f, 0.f) {}
+    inline constexpr PointXYZHSV (): PointXYZHSV (0.f, 0.f, 0.f) {}
 
     // @TODO: Use strong types??
     // This is a dangerous type, doesn't behave like others
-    inline PointXYZHSV (float _h, float _s, float _v):
+    inline constexpr PointXYZHSV (float _h, float _s, float _v):
       PointXYZHSV (0.f, 0.f, 0.f, _h, _s, _v) {}
 
-    inline PointXYZHSV (float _x, float _y, float _z,
-                        float _h, float _s, float _v)
-    {
-      x = _x; y = _y; z = _z;
-      data[3] = 1.0f;
-      h = _h; s = _s; v = _v;
-      data_c[3] = 0;
-    }
+    inline constexpr PointXYZHSV (float _x, float _y, float _z,
+                        float _h, float _s, float _v) :
+      _PointXYZHSV{{{_x, _y, _z, 1.0f}}, {{_h, _s, _v}}} {}
 
     friend std::ostream& operator << (std::ostream& os, const PointXYZHSV& p);
     PCL_MAKE_ALIGNED_OPERATOR_NEW
   };
 
-
-
-  PCL_EXPORTS std::ostream& operator << (std::ostream& os, const PointXY& p);
-  /** \brief A 2D point structure representing Euclidean xy coordinates.
-    * \ingroup common
-    */
-  struct PointXY
+  struct EIGEN_ALIGN16 _PointXY
   {
     union 
     { 
-      float data[2]; 
       struct 
       { 
         float x; 
         float y; 
       };
+      float data[4];
     };
+    PCL_MAKE_ALIGNED_OPERATOR_NEW
+  };
+
+
+  PCL_EXPORTS std::ostream& operator << (std::ostream& os, const PointXY& p);
+  /** \brief A 2D point structure representing Euclidean xy coordinates.
+    * \ingroup common
+    */
+  struct EIGEN_ALIGN16 PointXY : public _PointXY
```

Comment:
What was your reasoning for changing `PointXY` the way you did (introducing `_PointXY`, changing `float data[2]` to `float data[4]`, the alignment stuff, ...)? `PointXY` is used less often than the other types, and unless we are sure that the changes work well with the usual use cases, I would tend to leave it like it is (making the constructors `constexpr` is fine of course)

Reply:
yes, it was an overengineering, reverted these alignment-related changes